roadmap-voting-app/app/api/user/route.ts

```diff
@@ -0,0 +1,41 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { supabase } from '../../../lib/supabaseClient';
+
+export async function POST(req: NextRequest) {
+    try {
+        const login = req.nextUrl.searchParams.get("login") as string;
+        const formdata = await req.formData();
+        const email = formdata.get("email") as string;
+        const password = formdata.get("password") as string;
+        if (login) {
+            const { data , error } = await supabase.auth.signInWithPassword({ email, password });
```

Comment:
[maybe i am missing something]
where is the logic to sign up a new user? 
ref: https://supabase.com/docs/reference/javascript/auth-signup 

i see you have a sign in flow, but i don't see a way for a user to sign up here - can you help elaborate?

Reply:
looking at your screenshot, i think its might be coming as part of a different PR?

Reply:
so when the login parameter is provided then the logic is basically to sign in the user and if its not provided the logic is there which will sign up the user.

Line 28, 29:
            const signUpResponse = await supabase.auth.signUp({ email, password });
            if (signUpResponse.error) throw signUpResponse.error;